### PR TITLE
8283298: Make CodeCacheSegmentSize a product flag

### PR DESCRIPTION
**Pull request by DamonFool**
Hi all,

As discussed in https://github.com/openjdk/jdk/pull/7830, this patch makes `CodeCacheSegmentSize` a product flag.
It also fixes two bugs when testing the release VM with CodeEntryAlignment={512, 1024}.
Please review it.

Thanks.
Best regards,
Jie

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8283298](https://bugs.openjdk.java.net/browse/JDK-8283298): Make CodeCacheSegmentSize a product flag

### Reviewers
 * [Dean Long](https://openjdk.java.net/census#dlong) (@dean-long - **Reviewer**) ⚠️ Review applies to c84c659c288a167b941023870b0fc829cda2b9cc
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to af65cbaaca422cd451740eff65c5c8e68fcf4a05

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7851/head:pull/7851` \
`$ git checkout pull/7851`

Update a local copy of the PR: \
`$ git checkout pull/7851` \
`$ git pull https://git.openjdk.java.net/jdk pull/7851/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7851`

View PR using the GUI difftool: \
`$ git pr show -t 7851`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7851.diff">https://git.openjdk.java.net/jdk/pull/7851.diff</a>

</details>

### CODE REVIEW

**Review by dean-long — APPROVED** [review]
Looks good to me.

**Comment by DamonFool** [comment]
> Looks good to me.

Thanks @dean-long .

May I get a second review for this change?
Thanks.

File: src/hotspot/share/interpreter/templateInterpreter.cpp
Comment (vnkozlov) [inline src/hotspot/share/interpreter/templateInterpreter.cpp:51]: Please explain in comment where these numbers (280, *2) are coming from and why you need additional size.

File: src/hotspot/share/prims/methodHandles.cpp
Comment (vnkozlov) [inline src/hotspot/share/prims/methodHandles.cpp:93]: Add comment that we need additional bytes due to alignment.

**Comment by DamonFool** [comment]
Thanks @vnkozlov for the review.
The comments had been added in the code.
Thanks.

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by DamonFool** [comment]
> Good.

Thanks @vnkozlov .
I pushed one more commit to fix a comment typo.

**Pull request closed without merging**
